### ISSUE TITLE
Typing for clients/resources issue

### ISSUE THREAD

**Issue by Wurstnase** [youtype/mypy_boto3_builder#344]
**Describe the bug**
A `boto3.client` is always a `BaseClient`. Similar issue with `boto3.resource`.

**To Reproduce**

1. Install `boto3-stubs-lite[secretsmanager]==1.36.0`
2. Run `mypy` on the following code sample

```python
import boto3
from mypy_boto3_secretsmanager import SecretsManagerClient

def secretsmanager_client() -> SecretsManagerClient:
    return boto3.client("secretsmanager")
```

**Actual output**

```
error: Incompatible return value type (got "BaseClient", expected "SecretsManagerClient")  [return-value]
        return boto3.client("secretsmanager")
```

**Expected output**

no error expected

**Additional context**
MacOS
boto3==1.36.0

**Comment by vemel** [comment]
Hello.

This is expected. `boto3-stubs-lite` Does not add overloads for `boto3.client` methods, so return type is `BaseClient`.

You have three options:
- Replace with `boto3-stubs[secretsmanager]`. Probably you will have performance issues in PyCharm.
- Use `typing.cast`: `return cast(SecretsManagerClient, boto3.client("secretsmanager"))`
- Use `mypy_boto3_builder` directly to generate type annotations only for services you use. Run `mypy_boto3_builder` with no args (I recommend using `uvx`: `uvx mypy_boto3_builder`) and follow instructions. DO not forget to add `secretsmanager` to services you use.

Let me know if this resolves your issue.

**Comment by Wurstnase** [comment]
Hi vemel,

pretty sad, that this will not work anymore. For functions this may be a workaround.  
Why can't it be like in `<1.36.0`?

It is not very handy to use the lite stubs and needs casting anytime.

```python
sm_client: SecretsManagerClient = boto3.client("secretsmanager")
```

vs.

```python
sm_client = cast(SecretsManagerClient, boto3.client("secretsmanager"))
```

**Comment by Wurstnase** [comment]
A regression somehow?


<details>
<summary>v1.36.0</summary>

```
stubs v0.1.0
├── boto3 v1.36.0
│   ├── botocore v1.36.0
│   │   ├── jmespath v1.0.1
│   │   ├── python-dateutil v2.9.0.post0
│   │   │   └── six v1.17.0
│   │   └── urllib3 v2.3.0
│   ├── jmespath v1.0.1
│   └── s3transfer v0.11.0
│       └── botocore v1.36.0 (*)
├── boto3-stubs-lite[s3] v1.36.0
│   ├── botocore-stubs v1.36.0
│   │   └── types-awscrt v0.23.6
│   ├── types-s3transfer v0.11.0
│   └── mypy-boto3-s3 v1.36.0 (extra: s3)
└── mypy v1.14.1
    ├── mypy-extensions v1.0.0
    └── typing-extensions v4.12.2
```

</details>


![Image](https://github.com/user-attachments/assets/aed71a0f-2487-4259-a903-845fde950735)


<details>
<summary>v1.35.70</summary>

```
stubs v0.1.0
├── boto3 v1.36.0
│   ├── botocore v1.36.0
│   │   ├── jmespath v1.0.1
│   │   ├── python-dateutil v2.9.0.post0
│   │   │   └── six v1.17.0
│   │   └── urllib3 v2.3.0
│   ├── jmespath v1.0.1
│   └── s3transfer v0.11.0
│       └── botocore v1.36.0 (*)
├── boto3-stubs-lite[s3] v1.35.70
│   ├── botocore-stubs v1.36.0
│   │   └── types-awscrt v0.23.6
│   ├── types-s3transfer v0.11.0
│   └── mypy-boto3-s3 v1.35.93 (extra: s3)
└── mypy v1.14.1
    ├── mypy-extensions v1.0.0
    └── typing-extensions v4.12.2
```

</details>

![Image](https://github.com/user-attachments/assets/14def349-3513-4c5f-a544-7d93768c29ea)

**Comment by Wurstnase** [comment]
`boto3-stubs[s3]` has the same issue as the `lite` Version

<details>
<summary>boto3-stubs[s3] v1.36.0</summary>

stubs v0.1.0
├── boto3 v1.36.0
│   ├── botocore v1.36.0
│   │   ├── jmespath v1.0.1
│   │   ├── python-dateutil v2.9.0.post0
│   │   │   └── six v1.17.0
│   │   └── urllib3 v2.3.0
│   ├── jmespath v1.0.1
│   └── s3transfer v0.11.0
│       └── botocore v1.36.0 (*)
├── boto3-stubs[s3] v1.36.0
│   ├── botocore-stubs v1.36.0
│   │   └── types-awscrt v0.23.6
│   ├── types-s3transfer v0.11.0
│   └── mypy-boto3-s3 v1.36.0 (extra: s3)
└── mypy v1.14.1
    ├── mypy-extensions v1.0.0
    └── typing-extensions v4.12.2

</details>


![Image](https://github.com/user-attachments/assets/cb34d6cb-94d2-4f77-a8ad-f57c127b3e2f)

**Comment by vemel** [comment]
Indeed, it works in `boto3-stubs-lite==1.35.70`. Might be related to `boto3 1.36.0` release, because builder itself has not changed between these versions.

**Comment by vemel** [comment]
Fixed in `boto3-stubs-lite 1.36.0.post1` and `boto3-stubs-lite 1.36.1`.

Actually, due to this error I found many inconsistencies.

- Reverted `client/session` method return types to `Any`
- `boto3-stubs-lite` were not released on patch `boto3` update, so this issue appeared only on `boto3 1.36.0` minor release
- There were no integration tests for `*-lite` packages. I added tests for `boto3-stubs-lite` to catch issues like this one. I am going to add tests for `types-boto3-lite` and `types-botocore-lite` as well.

I did not release a new builder version, because the changes were mostly done in CI, and I also changed static files in https://github.com/youtype/types-boto3

**Comment by Wurstnase** [comment]
It works great.

Thank you very much. Another pretty quick fix.
